source/jormungandr/jormungandr/autocomplete/geocodejson.py

```diff
@@ -50,18 +50,51 @@ class GeocodeJson(AbstractAutocomplete):
     TYPE_STREET = "street"
 
     def __init__(self, **kwargs):
-        self.external_api = kwargs.get('host')
+        self.host = kwargs.get('host')
         self.timeout = kwargs.get('timeout', 10)
 
+    def call_bragi(self, url, method, **kwargs):
+        try:
+            return method(url, **kwargs)
+        except requests.Timeout:
+            logging.getLogger(__name__).error('autocomplete request timeout')
+            raise TechnicalError('external autocomplete service timeout')
+        except:
+            logging.getLogger(__name__).exception('error in autocomplete request')
+            raise TechnicalError('impossible to access external autocomplete service')
+
+    def response_marshaler(self, response_bragi):
+        json_response = response_bragi.json()
+        from flask.ext.restful import marshal
+        from jormungandr.interfaces.v1.Places import geocodejson
+
+        return marshal(json_response, geocodejson)
+
+    def make_url(self, end_point, uri=None):
+
+        if end_point not in ['autocomplete', 'features']:
+            raise TechnicalError('Unknown endpoint')
+
+        if not self.host:
+            raise TechnicalError('global autocomplete not configured')
+
+        url = "{host}/{end_point}".format(host=self.host, end_point=end_point)
+        if uri:
+            url = '{url}/{uri}'.format(url=url, uri=uri)
+        return url
+
+    def basic_params(self, instance):
```

Comment:
This method should return the common parameters to "get" and "get_uri", common_params ?